pkg/verify/verify.go

```diff
@@ -43,124 +52,78 @@ func (rejectVerifier) Verify(ctx context.Context, releaseDigest string) error {
-			continue
-		}
-		if _, ok := manifest.Obj.GetAnnotations()["release.openshift.io/verification-config-map"]; !ok {
-			continue
-		}
+// ClientBuilder provides a method for generating an HTTP Client configured
+// with cluster proxy settings, if they exist.
+type ClientBuilder interface {
+	// HTTPClient returns a client suitable for retrieving signatures. It is not
+	// required to be unique per call, but may be called concurrently.
+	HTTPClient() (*http.Client, error)
+}
 
-		src := fmt.Sprintf("the config map %s/%s", manifest.Obj.GetNamespace(), manifest.Obj.GetName())
-		data, _, err := unstructured.NestedStringMap(manifest.Obj.Object, "data")
-		if err != nil {
-			return nil, errors.Wrapf(err, "%s is not valid: %v", src, err)
-		}
-		verifiers := make(map[string]openpgp.EntityList)
-		var stores []*url.URL
-		for k, v := range data {
-			switch {
-			case strings.HasPrefix(k, "verifier-public-key-"):
-				keyring, err := loadArmoredOrUnarmoredGPGKeyRing([]byte(v))
-				if err != nil {
-					return nil, errors.Wrapf(err, "%s has an invalid key %q that must be a GPG public key: %v", src, k, err)
-				}
-				verifiers[k] = keyring
-			case strings.HasPrefix(k, "store-"):
-				v = strings.TrimSpace(v)
-				u, err := url.Parse(v)
-				if err != nil || (u.Scheme != "http" && u.Scheme != "https" && u.Scheme != "file") {
-					return nil, fmt.Errorf("%s has an invalid key %q: must be a valid URL with scheme file://, http://, or https://", src, k)
-				}
-				stores = append(stores, u)
-			default:
-				klog.Warningf("An unexpected key was found in %s and will be ignored (expected store-* or verifier-public-key-*): %s", src, k)
-			}
-		}
-		if len(stores) == 0 {
-			return nil, fmt.Errorf("%s did not provide any signature stores to read from and cannot be used", src)
-		}
-		if len(verifiers) == 0 {
-			return nil, fmt.Errorf("%s did not provide any GPG public keys to verify signatures from and cannot be used", src)
-		}
+// DefaultClient uses the default http.Client for accessing signatures.
+var DefaultClient = simpleClientBuilder{}
 
-		return &releaseVerifier{
-			verifiers:     verifiers,
-			stores:        stores,
-			clientBuilder: clientBuilder,
-		}, nil
-	}
-	return nil, nil
-}
+// simpleClientBuilder implements the ClientBuilder interface and may be used for testing.
+type simpleClientBuilder struct{}
 
-func loadArmoredOrUnarmoredGPGKeyRing(data []byte) (openpgp.EntityList, error) {
-	keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(data))
-	if err == nil {
-		return keyring, nil
-	}
-	return openpgp.ReadKeyRing(bytes.NewReader(data))
+// HTTPClient from simpleClientBuilder creates an http.Client with no configuration.
+func (s simpleClientBuilder) HTTPClient() (*http.Client, error) {
+	return &http.Client{}, nil
 }
 
+// maxSignatureSearch prevents unbounded recursion on malicious signature stores (if
+// an attacker was able to take ownership of the store to perform DoS on clusters).
 const maxSignatureSearch = 10
```

Comment:
this should be build into the store implementations, right?  By the time you get down to the verifier, the CVO has already been DoSed by the exploding store trying to fill the `DigestSignatures` request.